MCP_Server/grid_notation.py

```
@@ -0,0 +1,498 @@
+    if not notes:
+        return False
+
+    pitches = set(n.get('pitch', 60) for n in notes)
+    drum_pitches = sum(1 for p in pitches if 35 <= p <= 81)
+
+    # If most pitches are in drum range and there are multiple distinct pitches
+    # clustered in that range, it's likely drums
+    return drum_pitches >= len(pitches) * 0.8 and len(pitches) > 1
+
+
+def notes_to_grid(notes: list[dict], is_drums: bool = None, steps_per_beat: int = 4) -> str:
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)